components/brave_extension/extension/brave_extension/background/today/index.ts

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as Background from '../../../../../common/Background'
+import * as Feed from './feed'
+import * as Publishers from './publishers'
+import * as PublisherUserPrefs from './publisher-user-prefs'
+import { getUnpaddedAsDataUrl } from './privateCDN'
+
+const SETTINGS_KEY_SHOW_TODAY = 'brave.new_tab_page.show_brave_today'
+const ALARM_KEY_FEED_UPDATE = 'brave-today-update-feed'
+const ALARM_KEY_PUBLISHERS_UPDATE = 'brave-today-update-publishers'
+
+//
+// This module is the orchestrator for the Brave Today backend.
+// It handles command communication from front-ends, as well
+// as setting up and handling timers, and events from the
+// rest of the browser.
+//
```

Comment:
Nice comments here (and in other files too)! 😻 Definitely helps when reading this